### DISCUSSION TITLE
Downsides vs Zod

### DISCUSSION THREAD

**Discussion by sakgoyal** [open-circle/valibot#278] [General]
I've been using Zod and I really love it. its super intuitive.
But this seems like a no brainer, smaller, and has the same API?

So why wouldn't everyone use this? what are the downsides of using this over Zod?
Are there missing features or things I can't do with Valibot that I can with Zod?

**Comment by fabian-hiller** [comment]
Some features such as function and promise schemas are still missing in Valibot, but we plan to add this functionality in the coming weeks. The biggest difference is in implementation, API design, developer experience and ecosystem. With Zod, you can easily chain the methods which results in a great DX. With Valibot you have to import them one by one and add the functions to the right place. I wrote my bachelor thesis on this topic. Feel free to take a look: https://valibot.dev/thesis.pdf

**Reply by sakgoyal** [reply]
so is chaining something that is planned or not possible due to the way you have to import things individually?

**Reply by fabian-hiller** [reply]
Chaining like in Zod is not planned and not possible with our API design. But I have heard from many people who love the way Valibot's API is structured. Valibot feels familiar if you have worked with Zod before.
